pkg/crypto/tinkcrypto/primitive/composite/ecdh/ecdh_xchacha_aead_private_key_manager.go

```diff
@@ -0,0 +1,200 @@
+	tinkpb "github.com/google/tink/go/proto/tink_go_proto"
+
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite"
+	"github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/composite/ecdh/subtle"
+	ecdhpb "github.com/hyperledger/aries-framework-go/pkg/crypto/tinkcrypto/primitive/proto/ecdh_aead_go_proto"
+	"github.com/hyperledger/aries-framework-go/pkg/internal/cryptoutil"
+)
+
+const (
+	ecdhXChachaPrivateKeyVersion = 0
+	ecdhXChachaPrivateKeyTypeURL = "type.hyperledger.org/hyperledger.aries.crypto.tink.EcdhXChachaAeadPrivateKey"
+)
+
+// common errors.
+var (
+	errInvalidECDHXChachaPrivateKey       = errors.New("ecdh_xchacha_private_key_manager: invalid key")
+	errInvalidECDHXChachaPrivateKeyFormat = errors.New("ecdh_xchacha_private_key_manager: invalid key format")
+)
+
+// ecdhXChachaPrivateKeyManager is an implementation of PrivateKeyManager interface.
+// It generates new ECDHPrivateKey (XChacha) keys and produces new instances of ECDHAEADCompositeDecrypt subtle.
+type ecdhXChachaPrivateKeyManager struct{}
+
+// Assert that ecdhXChachaPrivateKeyManager implements the PrivateKeyManager interface.
+var _ registry.PrivateKeyManager = (*ecdhXChachaPrivateKeyManager)(nil)
+
+// newECDHXChachaPrivateKeyManager creates a new ecdhXChachaPrivateKeyManager.
+func newECDHXChachaPrivateKeyManager() *ecdhXChachaPrivateKeyManager {
+	return new(ecdhXChachaPrivateKeyManager)
+}
+
+// Primitive creates an ECDHESPrivateKey subtle for the given serialized ECDHESPrivateKey proto.
+func (km *ecdhXChachaPrivateKeyManager) Primitive(serializedKey []byte) (interface{}, error) {
+	if len(serializedKey) == 0 {
+		return nil, errInvalidECDHXChachaPrivateKey
+	}
+
+	key := new(ecdhpb.EcdhAeadPrivateKey)
+
+	err := proto.Unmarshal(serializedKey, key)
+	if err != nil {
+		return nil, errInvalidECDHXChachaPrivateKey
+	}
+
+	err = km.validateKey(key)
+	if err != nil {
+		return nil, errInvalidECDHXChachaPrivateKey
+	}
+
+	rEnc, err := composite.NewRegisterCompositeAEADEncHelper(key.PublicKey.Params.EncParams.AeadEnc)
+	if err != nil {
+		return nil, fmt.Errorf("ecdh_xchacha_private_key_manager: NewRegisterCompositeAEADEncHelper "+
+			"failed: %w", err)
+	}
+
+	return subtle.NewECDHAEADCompositeDecrypt(rEnc, key.PublicKey.Params.EncParams.CEK), nil
+}
+
+// NewKey creates a new key according to the specification of ECDHESPrivateKey format.
+func (km *ecdhXChachaPrivateKeyManager) NewKey(serializedKeyFormat []byte) (proto.Message, error) {
+	if len(serializedKeyFormat) == 0 {
+		return nil, errInvalidECDHXChachaPrivateKeyFormat
+	}
+
+	keyFormat := new(ecdhpb.EcdhAeadKeyFormat)
+
+	err := proto.Unmarshal(serializedKeyFormat, keyFormat)
+	if err != nil {
+		return nil, errInvalidECDHXChachaPrivateKeyFormat
+	}
+
+	err = validateKeyXChachaFormat(keyFormat.Params)
+	if err != nil {
+		return nil, errInvalidECDHXChachaPrivateKeyFormat
+	}
+
+	pub, pvt, err := ed25519.GenerateKey(rand.Reader)
```

Comment:
I don't understand the usage of ed25519.

Reply:
this function returns  key pair.

what is needed for key wrapping (ECDH) is a public/private key pair representing a recipient

Reply:
How do I use P curves with ChaCha? Why is the X25519 logic in this key manager instead of its own key manager?